pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter_test.go

```diff
@@ -23,6 +23,7 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"kubevirt.io/kubevirt/tests/libvmi"
```

Comment:
@iholder101 I am not sure if we want to include `tests` code in the production code.
I know there are (a few) other places where it is already done, but I want to make sure we want to proceed on this path.
IMO I will discourage this until the `libvmi` is included in the production code (or will have its own repo).
WDYT? Thanks

Reply:
So, TBH, I think that we made a big mistake placing `libvmi` inside the test package.

libvmi is about easily and declaratively creating VMIs or other virt related objects. It has nothing to do with tests specifically - the tests are just another user, same as production code. On the contrary - using our own package inside production code applies to the "[eating your own dog food](https://en.wikipedia.org/wiki/Eating_your_own_dog_food)" principle.

So yes - I think it's perfectly fine to use libvmi outside of the test package - and I think we should strive to extract libvmi from the test package altogether.

Reply:
However @fossedihelm I'm ready to be convinced otherwise if you think there's a good reason not to use it

Reply:
I agree with your reasoning, but I still think that it should be the test that uses the `libvmi` instead of the opposite, which also applies to the eating your own dog food principle.
From my POV the tests perform something on top of the production code, while with this we are allowing a bidirectional flow.
The fact that this bidirectional behavior was not present before reinforces my point of view.
IIRC there is an effort to move the `libvmi` inside the production code, so my question is:
Why don't we just first move it and then perform this kind of (great) improvement?
Having said that, if all the others agree with this I will for sure not proceed with a strong blocking behavior :slightly_smiling_face:

Reply:
Thank you @fossedihelm!

I've discussed this with @dankenigsberg, and after digging around I figured that there is almost no real dependency between libvmi and the tests, which is a pleasant surprise indeed! The only place where we depend on the tests, it seems, is the container-disk image resolution:
https://github.com/kubevirt/kubevirt/blob/ebd625c5fd03f5d7983d670281c41aa38ca84038/tests/libvmi/factory.go#L26
https://github.com/kubevirt/kubevirt/blob/ebd625c5fd03f5d7983d670281c41aa38ca84038/tests/libvmi/factory.go#L38-L43

I think that we can easily extract libvmi outside of `tests/` if we do the following:
1) Define a variable in libvmi with a default container disk image. This container disk image should be external and public and reside on quay.io or other public registry.
2) Introduce a `RegisterDefaultContainerDisk` that would provide the ability to override this default container disk.

This is very similar to what we already do with default options:
https://github.com/kubevirt/kubevirt/blob/ebd625c5fd03f5d7983d670281c41aa38ca84038/tests/libvmi/vmi.go#L49-L53

Of course we might need to have several images, one for each factory (fedora, alpine, etc).

@EdDev Any thoughts on this?

If we all agree, then I think I would make this PR a draft and instead make a PR to move libvmi to a different directory and package, outside of tests. Then I would undraft this PR again.

Reply:
Another option is to move the factory in its own file (in the test pkg)

Reply:
> @EdDev Any thoughts on this?
> Do you think we should wait with this PR until we move libvmi out of the tests, or do you think it's ok to work on both in parallel?

I posted the PR to promote libvmi under pkg. It leaves the factory at the e2e tests as they are not relevant to the rest of the usages. 
#11516

Reply:
Awesome @EdDev! Thank you! Let's discuss it in the PR!

Let's hold this PR until yours merges.
/hold